### PR TITLE
Added Delete Functionality

### PR DESCRIPTION
**Pull request by OlypsisAli**
Previously, users were unable to delete files via the estuary dashboard. They now have that ability.

### CODE REVIEW

**Comment by OlypsisAli** [comment]
@jimmylee Note that I decided to repull the latest code and reimplement my previous PR. Let me know if you have any feedback.

File: components/FilesTable.tsx [hunk 1]
```diff
@@ -66,6 +68,32 @@ const FilesTable = ({ files }) => {
         width: '30%',
         Filter: DefaultColumnFilter,
       },
+      {
+        Header: 'Delete',
+        accessor: (data) => String(data.id).padStart(9, '0'),
```
Comment (neelvirdy) [inline components/FilesTable.tsx:73]: can just return `data.id` here

File: components/FilesTable.tsx [hunk 2]
```diff
@@ -66,6 +68,32 @@ const FilesTable = ({ files }) => {
...
+              type="file"
+              style={{ backgroundColor: '#f54b42' }}
+            >
+              Delete
+            </Button>
+          </div>
+        ),
+        // accessor: (data) => data.aggregatedFiles + 1,
+        width: '9%',
+        maxWidth: '9%',
```
Comment (neelvirdy) [inline components/FilesTable.tsx:94]: maxWidth isn't needed

File: components/FilesTable.tsx [hunk 3]
```diff
@@ -2,9 +2,11 @@ import * as U from '@common/utilities';
 import tstyles from '@pages/files-table.module.scss';
 import React, { useMemo, useState } from 'react';
 import { useFilters, usePagination, useSortBy, useTable } from 'react-table';
+import Button from '@components/Button';
+import * as R from '@common/requests';
 
-const FilesTable = ({ files }) => {
-  const [gateway, setGateway] = useState('https://gateway.estuary.tech/gw/ipfs/');
+const FilesTable = ({ files, setFiles }) => {
```
Comment (neelvirdy) [inline components/FilesTable.tsx:8]: nit: `setFiles` -> `onDelete`

**Comment by neelvirdy** [comment]
Thanks for opening a PR @OlypsisAli. The complexity here is that many files are no longer safely deletable. If they have entered the dealmaking stage, we can't allow a delete because it would break dealmaking for that file and all other files aggregated with it.

The only case you can safely delete in is when the file is still in a staging bucket that has not been aggregated. Instead of putting the delete button here and returning errors in all other cases, can you move the delete button into the StagingTable?

**Review by gmelodie — CHANGES_REQUESTED** [review]
Tiny comment. Other than that I really like your PR!! Thank you for doing this :)

File: components/FilesTable.tsx [hunk 4]
```diff
@@ -2,9 +2,11 @@ import * as U from '@common/utilities';
...
+  const [gateway, setGateway] = useState('https://api.estuary.tech/gw/ipfs/');
```
[the remark this answers is not in the document]
Comment (alvin-reyes) [reply components/FilesTable.tsx:9]: yes the `gateway.estuary.tech` is peered with the API and shuttles so we can use it similar to how we use dweb and our api node so let's use `https://gateway.estuary.tech/gw/ipfs/` instead.

Some examples:

- https://gateway.estuary.tech/gw/ipfs/bafybeigvgzoolc3drupxhlevdp2ugqcrbcsqfmcek2zxiw5wctk3xjpjwy
- https://gateway.estuary.tech/gw/ipfs/QmPBHAjRLZqvJwcBUTiVxNtvugToAnTyJxpzTCgKZVHsvw

**Comment by neelvirdy** [comment]
> @neelvirdy this seems like an estuary problem, not a frontend one. I mean, nothing stops me from deleting a file while deal making is in process if I just call the API. I think just unpinning the CIDs is good enough in terms of deletion.

in dealmaking and reconciliation, we dont handle the cases where data has been deleted yet - so we just prevent it if we've entered those stages

so you can call the API but it will throw an error if it's a content that's been aggregated in a zone and that zone has started dealmaking. instead of setting up the user to hit the error pretty often in the frontend, i'd rather put it in a page (Staging) where all of the files are in a state that will not lead to an API error

**Comment by jimmylee** [comment]
Going to close this since its lingered for a while, feel free to re-open when you've had a chance to work through our comments.

**Pull request closed without merging**